### PR TITLE
Improve local testing documentation

### PR DESCRIPTION
**Pull request by JulianMaurin**
Hello :wave: 

According to the discussion on https://github.com/biopython/biopython/pull/3939, 
I'm updating the local testing documentation section. 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit``
locally, and understand that continuous integration checks will be used to
confirm the Biopython unit tests and style checks pass with these changes.

- [ ] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -76,16 +76,25 @@ this, run::
 Local Testing
 -------------
 
-Please always run the style checks (see above) and the full test suite on
-your local computer before submitting a pull request, e.g.::
+Please always run the full test suite locally before submitting a pull
+request, e.g.::
 
-    $ git commit Bio/XXX.py Tests/test_XXX.py  -m "Fixed bug 123"
```
Comment (JulianMaurin) [inline CONTRIBUTING.rst:82]: according to the last sentence, we expect people to commit after the execution of the test suite

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -76,16 +76,25 @@ this, run::
...
     $ python setup.py build
     $ python setup.py test
+    $ git commit ...
 
-If you have multiple versions of Python installed, ideally test them all
-(the Python tool ``tox`` can be helpful here).
```
[the remark this answers is not in the document]
Comment (peterjc) [reply CONTRIBUTING.rst:87]: The linked documentation does not talk about pytest or coverage though?

[the remark this answers is not in the document]
Comment (peterjc) [reply CONTRIBUTING.rst:87]: It is probably simplest to refer to the testing chapter of the tutorial for more details, and keep the CONTRIBUTING.rst file short and focused.

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -76,16 +76,25 @@ this, run::
...
+    $ pytest Tests/test_PDBList.py::TestPBDListGetList
+
+In order to generate the test coverage report, you can run the test suite
+using `coverage <https://coverage.readthedocs.io/>`_, eg.::
+
+    $ coverage run run_tests.py --offline
+    $ pytest --cov Bio/PDB/PDBList.py Tests/test_PDBList.py
+
+It would be possible to run the test suite via tox, but there is no
+configuration provided on the repository.
```
Comment (JulianMaurin) [inline CONTRIBUTING.rst:97]:
```suggestion
Have a look at the `related chapter <http://biopython.org/DIST/docs/tutorial/Tutorial.html#chapter%3Atesting>`_ in the documentation for more details.
```

File: CONTRIBUTING.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (JulianMaurin) [reply CONTRIBUTING.rst:87]: Ok, see the proposal at https://github.com/biopython/biopython/pull/4020/files#r941191916 ?

**Review by peterjc — APPROVED** [review]

**Review by peterjc — APPROVED** [review]
Assuming a squash-and-merge, this looks good to me.

**Pull request merged**